pkg/secrethub/auth_aws.go

```diff
@@ -0,0 +1,62 @@
+package secrethub
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/auth"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+)
+
+type awsAuthService struct {
+	client client
+}
+
+func newAWSAuthService(client client) AuthMethodService {
+	return &awsAuthService{
+		client: client,
+	}
+}
+
+func (s awsAuthService) Authenticate() (auth.Authenticator, error) {
+	// Currently always use the eu-west-1 region.
+	region := endpoints.EuWest1RegionID
+
+	cfg := aws.NewConfig().WithRegion(region).WithEndpoint("sts." + region + ".amazonaws.com")
```

Comment:
This looks kinda weird: you're passing region to it and then `WithEndpoint` in which you have to set the region again. If this is awkwardness is AWS's design then cool, if it's because we're using it wrong let's see how we can improve. 

I like their use of functional options though 😍 (although it could have even been better if it wasn't used on configs but on the actual client)

Reply:
It's a combination of awkwardness of the design in combination with using it weirdly. 

`.WithRegion(region)` is to signal to the AWS Client that it should sign it's requests for that region. It also changes the endpoint..... BUT specifically for `STS` the client refuses to use any other endpoint than `sts.amazonaws.com`, which is the _global_ STS endpoint (in the US). We like to use the local endpoints because the `eu-west-1` endpoint is closer to our server. Haven't found another way of achieving this than doing this workaround.

Reply:
Oh wow, that's interesting. There's actually a very good reason for doing it this awkwardly 🎉  Consider adding a small comment to explain what were doing here so we'll still know the rationale when someone comes along in 6 months and thinks "hey this is weird, it can be simpler".

Reply:
@SimonBarendse for you